### PR TITLE
fix: fix numbered list

### PR DESCRIPTION
**Pull request by redgeoff**
Hey @echo724, I just wanted to start off by saying thanks for maintaining this wonderful repo!

In Notion, I have discovered that a document with the following format:

```md
1. Thing1:
\t1. Thing2:
\t\t1. one
\t\t2. two
<br/>
Stuff
<br/>
1. Schema
\t1. id (integer)
\t2. type (string)
```

experiences issues with item numbering, which is incorrect and even changes unpredictably.

Upon debugging, I found that the use of concurrency with the ThreadPoolExecutor causes the shared state of the variables `_continued_numbered_list` and `_numbered_list_number` in the [block converter](https://github.com/mindfuldataai/notion2md/blob/1a633515ed835e9cda26cd0b5951773f2076a069/notion2md/convertor/block.py#L25-L26) to result in unpredictable and frequently incorrect numbering (a race condition).

To address this issue, I plan to use Test-Driven Development (TDD) to introduce a new test that expects the correct format shown above, and mock the Notion blocks for this test. Following this, I will submit the fix to the block converter and expect the test to pass.

### CODE REVIEW

**Comment by redgeoff** [comment]
I made the fixes to the block converter and the tests pass!

This PR should be all set to merge unless you see anything else that you'd like changed. Thanks @echo724!

**Comment by redgeoff** [comment]
Hey @echo724, is there anything else you'd like changed here?

**Comment by echo724** [comment]
I'm sorry for checking this so late. I've been swamped with another project recently. I will try to review as quickly as possible! Thank you for showing interest in Notion2MD!

File: notion2md/convertor/block.py [hunk 1]
```diff
@@ -22,41 +22,51 @@ def __init__(self, config: Config, client: NotionClient, io: IO = None):
...
         self._io = io
-        self._continued_numbered_list = False
-        self._numbered_list_number = 1
 
     def convert(self, blocks: dict) -> str:
         outcome_blocks: str = ""
         with concurrent.futures.ThreadPoolExecutor() as executor:
             results = executor.map(self.convert_block, blocks)
-            outcome_blocks = "".join([result for result in results])
+            outcome_blocks = "".join([result["text"] for result in results])
```
Comment (echo724) [inline notion2md/convertor/block.py:30]: That's an interesting solution. By resetting the number when parsing child blocks, you can ensure that indented blocks retain their sequence. My only concern is the change in the return method, which could potentially throw an unexpected exception somewhere. Have you ever run the application in a CLI environment besides StringExporter? If not, I suggest you try it out. If it runs well, then I can proceed with the merge.

File: tests/test_numbered_list.py
Comment (echo724) [inline tests/test_numbered_list.py:1]: That's a tremendous amount of mock responses. You must have worked hard to create them.

**Review by echo724 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**